Board.java

```diff
@@ -48,23 +55,21 @@ public boolean checkForWin(char player) {
         return (getWinner().equals(Character.toString(player)));
     }
 
-    public boolean containsOnlySame(char player, String row) {
-        for (int i=0; i < dimension; i++) {
-            if (!(row.charAt(i) == player)) return false;
-        }
-        return true;
+    public List<String> containsOnlySame(String player, List<String> combination) {
+        return combination.stream()
+                .filter(p -> p.contains(player))
+                .map(i -> i + 1)
+                .collect(toList());
     }
 
     public String getWinner() {
-        for (String combination : allWinningCombinations()) {
-            if (containsOnlySame('x', combination)) {
-                return "x";
-            }
-            if (containsOnlySame('o', combination)) {
-                return "o";
-            }
-        }
-        return "No winner";
+        final String[] winner = {"No winner"};
+
+        allWinningCombinations().forEach(combination -> {
+            if (containsOnlySame("o", combination).size() == dimension) winner[0] = "o";
+            if (containsOnlySame("x", combination).size() == dimension) winner[0] = "x";
```

Comment:
I know it would never be possible for both of these conditionals to be true, I still think an If(blah) else if (blah) would be more clear.